eth/downloader/peer.go

```diff
@@ -57,11 +60,10 @@ type peer struct {
 	blockIdle   int32 // Current block activity state of the peer (idle = 0, active = 1)
 	receiptIdle int32 // Current receipt activity state of the peer (idle = 0, active = 1)
 	stateIdle   int32 // Current node data activity state of the peer (idle = 0, active = 1)
-	rep         int32 // Simple peer reputation
 
-	blockCapacity   int32 // Number of blocks (bodies) allowed to fetch per request
-	receiptCapacity int32 // Number of receipts allowed to fetch per request
-	stateCapacity   int32 // Number of node data pieces allowed to fetch per request
+	blockThroughput   float64 // Number of blocks (bodies) measured to be retrievable per second
```

Comment:
@Gustav-Simonsson I also don't quite understand this aspect. I'd gladly rework it given the appropriate reasoning. One valid point I could see is that ints could be sorted out atomically vs. floats require a mutex, but even previously the atomic approach required a bit of mucking around to ensure concurrent updates all get processed and don't overwrite each other, so it was kind of already abusing atomic ops. This was it's a bit clearer and the mutex is already in there for maintaining the ignore list, so might as well have a tad cleaner code. Given that these are updated once per xecuted network request (i.e. encryption, networking, etc), it shouldn't matter from a performance point of view.